internal/resource/slicing.go

```diff
@@ -0,0 +1,121 @@
+package resource
```

Comment:
What do you think about the naming here? `resource.Slice` and `resource.New`

Reply:
In general, I like `New` when "main" type of a package is named after the package itself. If that's not the case or there's multiple types the `NewType` format, but that's just personal preference. I don't mind `Slice` :)

Reply:
I like it - renamed